provider/scaleway/scaleway.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2alpha2"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+const (
+	scalewyRecordTTL        uint32 = 300
+	scalewayDefaultPriority uint32 = 0
+	scalewayPriorityKey     string = "scw/priority"
+)
+
+// ScalewayProvider implements the DNS provider for Scaleway DNS
+type ScalewayProvider struct {
+	provider.BaseProvider
+	domainAPI DomainAPI
+	dryRun    bool
+	// only consider hosted zones managing domains ending in this suffix
+	domainFilter endpoint.DomainFilter
+}
+
+// ScalewayChange differentiates between ChangActions
+type ScalewayChange struct {
+	Action string
+	Record []domain.Record
+}
+
+// NewScalewayProvider initializes a new Scaleway DNS provider
+func NewScalewayProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*ScalewayProvider, error) {
+	accessKey, ok := os.LookupEnv(scw.ScwAccessKeyEnv)
```

Comment:
You could remove all this `LookupEnv` and check if the client is valid after calling `NewClient`